pkg/resources/secret_common.go

```diff
@@ -0,0 +1,112 @@
+package resources
+
+import (
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretCommonSchema = map[string]*schema.Schema{
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("String that specifies the identifier (i.e. name) for the secret, must be unique in your schema."),
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"database": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("The database in which to create the secret"),
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"schema": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("The schema in which to create the secret."),
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the secret.",
+	},
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW SECRET` for the given secret.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowSecretSchema,
+		},
+	},
+	DescribeOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `DESCRIBE SECRET` for the given secret.",
+		Elem: &schema.Resource{
+			Schema: schemas.DescribeSecretSchema,
+		},
+	},
+	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
+}
+
+type commonSecretCreate struct {
+	name     string
+	database string
+	schema   string
+	comment  *string
+}
+
+func handleSecretCreate(d *schema.ResourceData) commonSecretCreate {
```

Comment:
tbh, I don't feel this function gives much value, and it could be inlined. In `CreateContextSecretWithBasicAuthentication` you even forgot that the comment is read here.

Reply:
I have refactored it just to get the `database`, `schema` and `name` values. I also don't think its much of a use, if removing this would benefit in more readability please give me a comment on that on the re-review and it will be removed